sql/core/src/test/scala/org/apache/spark/sql/SQLQuerySuite.scala

```diff
@@ -2056,4 +2056,11 @@ class SQLQuerySuite extends QueryTest with SharedSQLContext {
       )
     }
   }
+
+  test("SPARK-13056: Null in map value causes NPE") {
+    val df = Seq(1 -> Map("abc" -> "somestring", "cba" -> null)).toDF("key", "value")
+    df.registerTempTable("maptest")
+    checkAnswer(sql("SELECT value['abc'] FROM maptest"), Row("somestring"))
+    checkAnswer(sql("SELECT value['cba'] FROM maptest"), Row(null))
```

Comment:
Have you verified if this throws NPE without this fix ? The test runs fine over trunk:

```
scala> val df = Seq(1 -> Map("abc" -> "somestring", "cba" -> null)).toDF("key", "value")
df: org.apache.spark.sql.DataFrame = [key: int, value: map<string,string>]

scala> df.registerTempTable("maptest")

scala> sqlContext.sql("SELECT value['cba'] FROM maptest").collect()
res28: Array[org.apache.spark.sql.Row] = Array([null])

scala> sqlContext.sql("SELECT value['cba'] FROM maptest").foreach(println)
[null]
```